BitwardenShared/Core/Auth/Services/AuthService.swift

```diff
@@ -704,6 +729,19 @@ class DefaultAuthService: AuthService { // swiftlint:disable:this type_body_leng
 
     // MARK: Private Methods
 
+    /// Check whether the user can unlock their vault with a Key Connector key.
+    ///
+    /// - Parameter response: The response received from the identity token request.
+    /// - Returns: Whether the vault can be unlocked with a Key Connector key.
+    ///
+    private func canUnlockWithKeyConnectorKey(_ response: IdentityTokenResponseModel) async throws -> Bool {
+        guard response.keyConnectorUrl != nil ||
+            response.userDecryptionOptions?.keyConnectorOption?.keyConnectorUrl != nil
```

Comment:
🎨  In [MAUI code](https://github.com/bitwarden/mobile/blob/main/src/Core/Services/AuthService.cs#L552) we're also checking that this url is not empty.